amazon-redshift-plugin/src/test/java/io/cdap/plugin/amazon/redshift/RedshiftSourceTestRun.java

```diff
@@ -0,0 +1,329 @@
+      "\"TEXT_COL\", \"DOUBLE_PREC_COL\" FROM my_table " +
+      "WHERE \"ID\" < 3 AND $CONDITIONS";
+    String boundingQuery = "SELECT MIN(\"ID\"),MAX(\"ID\") from my_table";
+    String splitBy = "ID";
+    ETLPlugin sourceConfig = new ETLPlugin(
+      RedshiftConstants.PLUGIN_NAME,
+      BatchSource.PLUGIN_TYPE,
+      ImmutableMap.<String, String>builder()
+        .putAll(BASE_PROPS)
+        .put(AbstractDBSource.DBSourceConfig.IMPORT_QUERY, importQuery)
+        .put(AbstractDBSource.DBSourceConfig.BOUNDING_QUERY, boundingQuery)
+        .put(AbstractDBSource.DBSourceConfig.SPLIT_BY, splitBy)
+        .put(Constants.Reference.REFERENCE_NAME, "DBSourceTest")
+        .build(),
+      null
+    );
+
+    String outputDatasetName = "output-dbsourcetest";
+    ETLPlugin sinkConfig = MockSink.getPlugin(outputDatasetName);
+
+    ApplicationManager appManager = deployETL(sourceConfig, sinkConfig,
+                                              DATAPIPELINE_ARTIFACT, "testDBSource");
+    runETLOnce(appManager);
+
+    DataSetManager<Table> outputManager = getDataset(outputDatasetName);
+    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
+
+    Assert.assertEquals(2, outputRecords.size());
+    String userid = outputRecords.get(0).get("NAME");
+    StructuredRecord row1 = "user1".equals(userid) ? outputRecords.get(0) : outputRecords.get(1);
+    StructuredRecord row2 = "user1".equals(userid) ? outputRecords.get(1) : outputRecords.get(0);
+
+    // Verify data
+    Assert.assertEquals("user1", row1.get("NAME"));
+    Assert.assertEquals("user2", row2.get("NAME"));
+    Assert.assertEquals("user1", row1.get("TEXT_COL"));
+    Assert.assertEquals("user2", row2.get("TEXT_COL"));
+    Assert.assertEquals("char1", ((String) row1.get("CHAR_COL")).trim());
+    Assert.assertEquals("char2", ((String) row2.get("CHAR_COL")).trim());
+    Assert.assertEquals(124.45f, ((Float) row1.get("SCORE")).doubleValue(), 0.000001);
+    Assert.assertEquals(125.45f, ((Float) row2.get("SCORE")).doubleValue(), 0.000001);
+    Assert.assertEquals(false, row1.get("GRADUATED"));
+    Assert.assertEquals(true, row2.get("GRADUATED"));
+    Assert.assertNull(row1.get("NOT_IMPORTED"));
+    Assert.assertNull(row2.get("NOT_IMPORTED"));
+
+    Assert.assertEquals(1, (int) row1.get("SMALLINT_COL"));
+    Assert.assertEquals(2, (int) row2.get("SMALLINT_COL"));
+    Assert.assertEquals(1, (long) row1.get("BIG"));
+    Assert.assertEquals(2, (long) row2.get("BIG"));
+
+    Assert.assertEquals(new BigDecimal(124.45, new MathContext(PRECISION)).setScale(SCALE),
+                        row1.getDecimal("NUMERIC_COL"));
+    Assert.assertEquals(new BigDecimal(125.45, new MathContext(PRECISION)).setScale(SCALE),
+                        row2.getDecimal("NUMERIC_COL"));
+    Assert.assertEquals(new BigDecimal(124.45, new MathContext(PRECISION)).setScale(SCALE),
+                        row1.getDecimal("DECIMAL_COL"));
```

Comment:
Same suggestion for BigDecimal as above.
```suggestion
    Assert.assertEquals(new BigDecimal("124.45", new MathContext(PRECISION)).setScale(SCALE),
                        row1.getDecimal("NUMERIC_COL"));
    Assert.assertEquals(new BigDecimal("125.45", new MathContext(PRECISION)).setScale(SCALE),
                        row2.getDecimal("NUMERIC_COL"));
    Assert.assertEquals(new BigDecimal("124.45", new MathContext(PRECISION)).setScale(SCALE),
                        row1.getDecimal("DECIMAL_COL"));
```